### ISSUE TITLE
Remove address fields for free checkout

### ISSUE THREAD

**Issue by helgatheviking** [woothemes/woocommerce#42496]
## Is your feature request related to a problem? Please describe.

I have a [plugin that removes the address fields for free checkouts](https://wordpress.org/plugins/remove-woocommerce-billing-address-for-free-checkout/). 

## Describe the solution you'd like
Wondering if there's any way to introduce parity with the shortcode checkout and do that same thing of conditionally removing fields.

## Describe alternatives you've considered
As far as I can tell  [here](https://github.com/woocommerce/woocommerce-blocks/blob/c69f13d7e6ab1b1ef7ba528e7b7015d6aacea179/src/StoreApi/Schemas/V1/BillingAddressSchema.php#L31) and in the [abstract](https://github.com/woocommerce/woocommerce-blocks/blob/95e460409816f5dcd896cd81c5c7ab100ef5f817/src/StoreApi/Schemas/V1/AbstractAddressSchema.php#L16). I don't see any filters or any of the more familiar WP modes of changing the result of `get_properties()`

**Comment by opr** [comment]
@helgatheviking thank you for bringing this issue up. Right now there's no alternative or workaround that I can suggest, however, I will add it to the backlog of issues so our leads can be aware of it when planning work.

**Comment by BrianHenryIE** [comment]
An example where I modified the checkout: I moved the email field to the beginning and added a login form inline.

![WooCommerce Checkout](https://github.com/BrianHenryIE/bh-wp-account-and-login-ux/blob/master/assets/bh-wp-account-and-login-ux-checkout.gif?raw=true "WooCommerce checkout login changes")

https://github.com/BrianHenryIE/bh-wp-account-and-login-ux

I haven't tried porting it to the new checkout, but from what I understand it is not possible.

**Comment by pierorocca** [comment]
Hi @helgatheviking - thanks for the suggestion.

I know this has been open for a very long time, and I appreciate your patience. We're currently going through some older WooCommerce issues, identifying what still needs fixing, and tidying up what's already fixed or out of scope.

I've reviewed this from a product perspective, and I'm already looking at billing address closely and how a checkout refactor might make them conditional based on if the payment method even requires it - most don't. And to also cover use cases like free products as you've brought up here.

Long story short, when. this does show up it'll be baked into core block checkout making a plugin unnecessary.

Best.
